### DISCUSSION TITLE
Message needs to be a string or object with property 'text'

### DISCUSSION THREAD

**Discussion by xcapri** [kirimwa-id/forum#21] [Tanya Jawab] [answered]
Saya pakai laravel 8 dengan script seperti dibawah , saat menggunakan api dari kirimwa mendapati error ***BaileysError: message needs to be a string or object with property 'text'***

view saya : 

```
{{ $test }}
```

saya punya controller : 

                $pesan = view('p.pesan')->with('test', 'test view')->with('test', 'test lagi');
                		
                $response = Http::withHeaders([
        			'Authorization' => 'Bearer ***************',
        			'Content-Type' => 'application/json',
        		])->post('https://api.kirimwa.id/v1/messages', [
        			'phone_number' => $user->wa,
        			'message' => $pesan,
        			'device_id' => 'device-user',
        			'message_type' => 'text',
        		]);

                $this->info($response->body());

dapat respon :
```
{"id":"kwid-********************","status":"pending","message":"Message is pending and waiting to be processed.","meta":{"location":"https://api.kirimwa.id/v1/messages/kwid-*****************"}}
```
kemudian manual cek status :
```
curl 'https://api.kirimwa.id/v1/messages/kwid-********************' \
  -H "Authorization: Bearer -********************'" 
```

mendapat respon :
```

{
    "id": "kwid-********************",
    "message": "BaileysError: message needs to be a string or object with property 'text'",
    "status": "fail",
    "created_at": "2021-08-13T12:54:18.491Z",
    "payload": {
        "message": {},
        "phone_number": "62**********",
        "device_id": "***********",
        "message_type": "text",
        "caption": null,
        "is_group_message": false
    }
}

```

Jadi apakah memng tidak support dari API nya atau saya ada yang salah dalam penulisan code pada controller ?

Thanks.

**Comment by rioastamal** [comment, marked as answer]
Hao @xcapri,

Isi dari variabel `$pesan` seperti apa? Karena dari `payload`-nya adalah empty object `{}`

**Reply by xcapri** [reply]
Halo @rioastamal 

asumsikan isi ```$pesan``` 
seperti ini dalam controller email 
```                 
                $pesan = view('pesan')
                                    ->with(['test' => $this->user
                               ]);
```

isi view/pesan.blade.php :
```
test {{$test}}

```
sebelumnya saya test view untuk mengirim email berhasil , dan mendapatkan respon pada email : 

```
test {"id":1,"username":"admin",..............................danseterusnya}

```
![image](https://user-images.githubusercontent.com/43540712/129430167-29287863-9c71-4f84-abc9-4aca6bc557cf.png)


untuk sekerang saya coba samakan variable $pesan saya gunakan juga dalam isi ``` 'message' => $pesan ``` , pesan berhasil terkirim, namun bukan isi detail user seperti pada email diatas tapi hanya ``` [object Object] ```  
![image](https://user-images.githubusercontent.com/43540712/129430120-44e4acd8-0179-4c38-8c8e-6cc23e40e41b.png)


Terimakasih sebelumnya

**Reply by rioastamal** [reply]
Sebelum dikirim coba encode pesan tersebut sebagai JSON string.

```php
    			'phone_number' => $user->wa,
    			'message' => json_encode($pesan),
```

**Reply by xcapri** [reply]
solved thanks mas
